### PR TITLE
Remove first/last restriction on <figcaption> within <figure>

### PR DESCRIPTION
**Pull request by travisleithead**
See issue #177

### CODE REVIEW

**Comment by LJWatson** [comment]
@travisleithead
Changes look good to me.

Have rounded up test results. Seems we have ample interop for this change.
http://ljwatson.github.io/test-cases/figcaption/case.html

File: sections/semantics-grouping-content.include [hunk 1]
```diff
@@ -1358,9 +1358,7 @@
     <dt><a>Contexts in which this element can be used</a>:</dt>
     <dd>Where <a>flow content</a> is expected.</dd>
     <dt><a>Content model</a>:</dt>
-    <dd>Either: One <{figcaption}> element followed by <a>flow content</a>.</dd>
-    <dd>Or: <a>Flow content</a> followed by one <{figcaption}> element.</dd>
-    <dd>Or: <a>Flow content</a>.</dd>
+    <dd><a>Flow content</a> optionally with one <{figcaption}> element.</dd>
```
Comment (chaals) [inline sections/semantics-grouping-content.include]: Please change "with one" to "including" and "element" to "elements".

File: sections/semantics-grouping-content.include [hunk 2]
```diff
@@ -1400,7 +1398,7 @@
     meaning.
   </div>
 
-  The <span class="impl">first</span> <{figcaption}> element child of the element, if any,
+  Any <{figcaption}> element child of <{figure}>, if any,
```
Comment (chaals) [inline sections/semantics-grouping-content.include]: Please change the first "Any" to "The", and "element child" to "descendants".

File: sections/semantics-grouping-content.include [hunk 3]
```diff
@@ -1400,7 +1398,7 @@
...
   represents the caption of the <{figure}> element's contents. If there is no child
```
Comment (chaals) [inline sections/semantics-grouping-content.include]: s/represents/represent/

File: sections/semantics-grouping-content.include [hunk 4]
```diff
@@ -1562,7 +1560,7 @@
     <dt><a>Categories</a>:</dt>
     <dd>None.</dd>
     <dt><a>Contexts in which this element can be used</a>:</dt>
-    <dd>As the first or last child of a <{figure}> element.</dd>
+    <dd>As a child of a <{figure}> element.</dd>
```
Comment (chaals) [inline sections/semantics-grouping-content.include]: s/child/descendant/

**Comment by stevefaulkner** [comment]
> For the minimal testing I've done, the caption is the sum of all figcaption children,

This is not the case in firefox or chrome on windows and is not what is specfied in the HTML AAM (though this may need tightening up to ensure its crystal clear). In the case of the browsers cited the accessible name for the figure is derived from the first figcaption only (code example of figure with 2 figcaptions http://s.codepen.io/stevef/debug/xVjMKZ).

I don't see a good use case for multiple figcaptions and what it may do is make the figure accessible name overly verbose and nonsensical (if captions are referring to different things). The advice for such cases is to nest figures: http://www.html5accessibility.com/tests/figures-nested.html

File: sections/semantics-grouping-content.include (hunk 1 above)
Comment (travisleithead) [inline sections/semantics-grouping-content.include]: Done.

File: sections/semantics-grouping-content.include (hunk 2 above)
Comment (travisleithead) [inline sections/semantics-grouping-content.include]: Got it.

File: sections/semantics-grouping-content.include (hunk 3 above)
Comment (travisleithead) [inline sections/semantics-grouping-content.include]: OK.

File: sections/semantics-grouping-content.include (hunk 4 above)
Comment (travisleithead) [inline sections/semantics-grouping-content.include]: K.

**Comment by travisleithead** [comment]
Updated to include Chaals feedback. @stevefaulkner not sure what to do with your comment. Thoughts?

**Comment by stevefaulkner** [comment]
@travisleithead > not sure what to do with your comment. Thoughts?
I disagree that multiple figcaptions should be conforming, so don't want the commit merged until I have had time to consider further.

**Comment by ZoeBijl** [comment]
If I may weight in, what is the use case for removing this restriction? Is there a dire need for multiple `<figcaptions>`? It feels like it would overcomplicate figures.

**Comment by travisleithead** [comment]
The restriction was that `<figcaption>` must be the first or last element in a `<figure>`. We've seen that to match reality this restriction can be lifted. I think that restriction (at a minimum) should be lifted to match reality. However, I'm not sure about making multiple `<figcaption>`s conforming is the way to go. Looking forward to @stevefaulkner thoughts after he's taken time to consider :)

**Comment by ZoeBijl** [comment]
@travisleithead, okay, if it matches reality better; that makes sense. But multiple `<figcaption>`'s sounds like a bad idea to me.

**Comment by stevefaulkner** [comment]
I have no issue with lifting restriction of `<figcaption>` placement in `<figure>` and agree it is good to match reality. In regards to multiple `<figcaption>` this does not match reality on windows(chrome, firefox) and linux (firefox), (will check mac, but suspect it is the same) and does not match the [defined naming calculation](https://w3c.github.io/aria/html-aam/html-aam.html#figure-and-figcaption-elements) for figure/figcaption. I would strongly suggest, without any obvious advantages being documented, we keep the current author restriction on use of multiple `<figcaption>` and instead provide author advice about use of nested `<figure>` elements.  This method preserves a clear programmatic relationship for `<figcaption>` as the source of caption information for all child content of a `<figure>` element. I will create an issue to add advice around use of nested figures, to the html spec.

**Comment by travisleithead** [comment]
Now, we specify one `<figcaption>` per `<figure>` regardless of child position. (essentially a revert, but with some other fixup I found nearby).

**Comment by travisleithead** [comment]
@stevefaulkner are you OK with merging the current PR?

**Comment by stevefaulkner** [comment]
@travisleithead LGTM thanks.

**Pull request merged**